Server/User/routehelper.py

```diff
@@ -12,6 +12,34 @@
     which will detail the success or failure of the command
     as well as any other necessary information regarding the command.
 """
+
+# Login either with username/password or token
+def login(data):
+	username = None
+	pw = None
+	token = None
+	user = None
+
+	if "username" in data and "pw" in data:
+		username = data["username"]
+		password = data["pw"]
+
+		user = Users.objects.filter(username=username).first()
+		if user:
+			verified = User.userhelper.verifyPassword(password, user.password)
+			if verified:
+				return {"Success": True, "Token": User.userhelper.generateLoginToken(user), "Username": user.username}
+
+	elif "token" in data:
+		token = data['token']
+		user = Users.objects.filter(token=token).first()
+
+		if user:
+			return {"Success": True, "Token": token, "Username": user.username}
+		
+	return {"Success": False, "Error": "Invalid Username/Password"}
```

Comment:
So this message gets displayed if somehow an internal error occurred in which one of username, pw, or token is missing?  Seems like we want a different message there.  I see the "if" falls through to here if the user/pass is invalid, but what if there's no user, pass, or token in the data?